src/toil_scripts/gatk_germline/germline.py

```diff
@@ -1,304 +1,500 @@
+    # Determine extension on input file
+    base, ext1 = os.path.splitext(url)
+    _, ext2 = os.path.splitext(base)
+
+    # Produce or download BAM file
+    if {ext1, ext2} & {'.fq', '.fastq'}:
+        if not config.run_bwa:
+            raise ValueError('Not configured to run BWA. Please set run-bwa: True')
+        elif not rg_line:
+            raise ValueError('FASTQ file requires an RG line for BWA alignment')
+        else:
+            get_bam = job.wrapJobFn(setup_and_run_bwa_kit, url, config).encapsulate()
+
+    elif {ext1, ext2} & {'.bam'}:
+        job.fileStore.logToMaster("Downloading BAM: %s" % uuid)
+        get_bam = job.wrapJobFn(download_url_job,
+                                url,
+                                name='toil.bam',
+                                synapse_login=config.synapse_login,
+                                s3_key_path=config.ssec,
+                                disk=config.file_size).encapsulate()
+    else:
+        raise ValueError('Accepted formats: FASTQ or BAM\n'
+                         'Could not determine {} file type from URL:\n{}'.format(uuid, url))
+
+    rm_secondary = job.wrapJobFn(run_samtools_view,
+                                 get_bam.rv(),
+                                 flag='0x800',
+                                 ncores=config.cores,
+                                 disk=config.file_size, cores=config.cores)
+
+    sort_bam_disk = PromisedRequirement(lambda x: 3 * x.size + human2bytes('10G'),
+                                        rm_secondary.rv())
+    sort_bam = job.wrapJobFn(run_samtools_sort,
+                             rm_secondary.rv(),
+                             ncores=config.cores,
+                             cores=config.cores,
+                             disk=sort_bam_disk)
+
+    index_bam_disk = PromisedRequirement(lambda x: int(1.5 * x.size), sort_bam.rv())
+    index_bam = job.wrapJobFn(run_samtools_index, sort_bam.rv(), disk=index_bam_disk)
+
+    job.addChild(get_bam)
+    get_bam.addChild(rm_secondary)
+    rm_secondary.addChild(sort_bam)
+    sort_bam.addChild(index_bam)
+    return sort_bam.rv(), index_bam.rv()
+
+
+def setup_and_run_bwa_kit(job, url, config):
+    """
+    Downloads and aligns paired FASTQs to reference genome using BWA.
+
+    :param JobFunctionWrapping Job: passed by Toil
+    :param str url: FASTQ sample URL
+    :param Namespace config: Input parameters and shared FileStoreIDs
+    :return: BAM FileStoreID
+    :rtype: str
+    """
+    job.fileStore.logToMaster("Aligning sample: {}".format(config.uuid))
+    cores = multiprocessing.cpu_count()
+    # Download fastq files
+    fq1 = job.addChildJobFn(download_url_job,
+                            url,
+                            name='toil.1.fq',
+                            synapse_login=config.synapse_login,
+                            s3_key_path=config.ssec,
+                            disk=config.file_size)
+    # Assumes second fastq url is identical
```

Comment:
This is going to give someone the business in the future. Can we validate the URL prior to attempting the download? If it doesn't exist, we should throw a meaningful error.